service/controller/v2/resource/chart/desired.go

```diff
@@ -16,17 +24,48 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 
 	name := key.ChartName(customObject)
 	channel := key.ChannelName(customObject)
+	chartValues, err := r.getConfigMapValues(ctx, customObject)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
 	releaseVersion, err := r.apprClient.GetReleaseVersion(name, channel)
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
 
 	chartState := &ChartState{
 		ChartName:      key.ChartName(customObject),
+		ChartValues:    chartValues,
 		ChannelName:    key.ChannelName(customObject),
 		ReleaseName:    key.ReleaseName(customObject),
 		ReleaseVersion: releaseVersion,
 	}
 
 	return chartState, nil
 }
+
+func (r *Resource) getConfigMapValues(ctx context.Context, customObject v1alpha1.ChartConfig) (map[string]interface{}, error) {
+	chartValues := make(map[string]interface{})
```

Comment:
Chart values are read from the config map. They are stored in JSON format so they can be converted to a map[string]interface. We also need to pass JSON to Helm when installing or updating charts.